app/launcher.ts

```diff
@@ -0,0 +1,206 @@
+/* eslint-disable no-invalid-this */
+/* eslint-disable @typescript-eslint/naming-convention */
+import { IProgram } from "gatsby/internal"
+import { readJSON } from "fs-extra"
+import {
+  fork,
+  ChildProcessPromise as ChildProcess,
+  Output,
+} from "promisify-child-process"
+import type { PackageJson } from "gatsby"
+import path from "path"
+import detectPort from "detect-port"
+import fixPath from "fix-path"
+import {
+  LogObject,
+  StructuredEventType,
+  GlobalStatus,
+  Status,
+  IPCMessageType,
+  IPCMessage,
+} from "./ipc-types"
+
+fixPath()
+
+// Gatsby package type is missing these
+type PackageJsonWithBin = PackageJson & { bin: { gatsby: string } }
+
+async function getPackageJson(root: string): Promise<PackageJsonWithBin> {
+  return readJSON(`${root}/node_modules/gatsby/package.json`)
```

Comment:
You don't need a special function to readJSON. Can't you just call `require` on it?

Reply:
I don't do that because `require` is cached, and package.json may have been edited.

Reply:
You're a smart dude 👍